packages/hooks/use-in-view.ts

```diff
@@ -0,0 +1,79 @@
+import { useState, useEffect, useRef, Dispatch, SetStateAction } from "react";
+
+interface IntersectionObserverCallback {
+  (entries: IntersectionObserverEntry[]);
+}
+interface IntersectionObserverCallbackCreator {
+  (options: { onlyOnce: boolean }): IntersectionObserverCallback;
+}
+
+interface Options {
+  rootMargin?: string;
+  onlyOnce?: boolean;
+}
+
+interface UseInView {
+  (options?: Options): [boolean, React.MutableRefObject<undefined>];
+}
+
+// This is an array with all the `setIntersected` functions
+// created by `useInView`, so we can find the corresponding
+// one to each entry on the IntersectionObserver callback.
+let setFunctions: [
+  Dispatch<SetStateAction<boolean>>,
+  React.MutableRefObject<undefined>
+][] = [];
+
+// This observer will be used during the whole life of the process.
+let observer: IntersectionObserver;
+
+// This callback can be called with more than one entry,
+// so we need to filter them and call the corresponding
+// `setIntersected` function for each changed entry.
+let createCallback: IntersectionObserverCallbackCreator = options => {
+  return entries => {
+    const changed = entries.filter(entry => entry.isIntersecting);
```

Comment:
If we are considering just those entries that describe a transition from not-intersecting to intersecting, then, when the state of the hook changes to `true`, it will never be `false` again, right?

If that is the case, I do not understand very well what the `onlyOne` option is for. Uhm 🤔

Reply:
Yeah, this is a major mistake. I'll change this.

Reply:
Fixed